dash.el

```diff
@@ -2154,6 +2154,23 @@ N is the length of the returned list."
         (push (funcall fun (car r)) r))
       (nreverse r))))
 
+(defun -iota (count &optional start step)
+  "Return a list containing COUNT numbers.
+Starts from START and adds STEP each time.  The default START is
+zero, the default STEP is 1.
+This function takes its name from the corresponding primitive in
+the APL language."
+  (if (or (not (integerp count)) (< count 0))
+      (signal 'wrong-type-argument count))
+  (if (and step (zerop step)) (make-list count start)
```

Comment:
What if `start` is `nil`? According to the docstring, I don't think this is right:

```el
(-iota 2 nil 0) ; => (nil nil)
```

Reply:
Thanks, good catch--will have to fix that.

Reply:
I think it should be as simple as hoisting

```el
(or start (setq start 0))
(or step (setq step 1))
```
before the `zerop` check.

Reply:
Agreed, I think that's the simplest way to address it.